src/vmq_queue.erl

```diff
@@ -229,6 +211,22 @@ wait_for_offline(Event, State) ->
 
 wait_for_offline({set_last_waiting_acks, WAcks}, _From, State) ->
     {reply, ok, wait_for_offline, handle_waiting_acks_and_msgs(WAcks, State)};
+wait_for_offline({add_session, SessionPid, Opts}, From,
+                 #state{waiting_call={migrate, _OtherQueue, MigrationFrom}} = State) ->
+    %% very racy...
```

Comment:
It's not clear to me what exactly is racy here. Can you give an example of what it is that will race and what the implications would be? Or is it OK that it is racy since the effect can be ignored?